arch/arm/soc/st_stm32/stm32f4/arm_mpu_mem_cfg.h

```diff
@@ -20,6 +20,8 @@
 #define REGION_FLASH_SIZE REGION_512K
 #elif CONFIG_FLASH_SIZE == 1024
 #define REGION_FLASH_SIZE REGION_1M
+#elif CONFIG_FLASH_SIZE == 1536
+#define REGION_FLASH_SIZE REGION_1M
```

Comment:
I agree. This matches my second suggestion, but without messing with subregions. I will do the change.